packages/vaex-ml/vaex/ml/sklearn.py

```diff
@@ -95,3 +95,141 @@ def fit(self, df, target, **kwargs):
         X = df[self.features].values
         y = df.evaluate(target)
         self.model.fit(X=X, y=y, **kwargs)
+
+
+@vaex.serialize.register
+@generate.register
+class IncrementalPredictor(state.HasState):
+    '''This class wraps any scikit-learn estimator (a.k.a predictions) that has
+    a `.partial_fit` method, and makes it a vaex pipeline object.pipeline
+
+    By wrapping "on-line" scikit-learn estimators with this class, they becoes a vaex
+    pipeline object. Thus, they can take full advantage of the serialization and
+    pipeline system of vaex. While the underlying estimator need to call the
+    `.partial_fit` method, this class contains the standard `.fit` method, and
+    the rest happens behind the scenes. One can also iterate over the data
+    multiple times (epochs), and optionally shuffle each batch before it is sent
+    to the estimator. The `predict` method returns a numpy, while the `transform`
+    method adds the prediction as a virtual column to a vaex DataFrame.
+
+    Note that the `.fit` method will use as much memory as needed to copy one
+    batch of data, while the `.predict` method makes a full memory copy.
+    The `transform` method is evaluated lazily, and no memory copies are made.
+
+    The scikit-learn estimators themselves are not modified at all, they are
+    taken from your local installation of scikit-learn.
+
```

Comment:
I think the wording is a bit odd, maybe we can completely remove it, or rephrase it simpler. "Note, we are using normal sklearn without modifications here."

Reply:
I just copy-pasted it from the regular SKLearnPredictor, but we can change the wording